tests/tools/fswalker/reporter/reporter_test.go

```diff
@@ -0,0 +1,134 @@
+package reporter
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+
+	"github.com/golang/protobuf/ptypes/timestamp"
+	fspb "github.com/google/fswalker/proto/fswalker"
+
+	"github.com/kopia/kopia/tests/testenv"
+)
+
+func TestReporterWithFiles(t *testing.T) {
+	ctx := context.TODO()
+
+	config := &fspb.ReportConfig{
+		Version:    1,
+		ExcludePfx: nil,
+	}
+
+	fileList := []*fspb.File{
+		{
+			Version: 0,
+			Path:    filepath.Join("some", "path"),
+			Info: &fspb.FileInfo{
+				Name: "this_is_a.file",
+				Size: 11235,
+				Mode: 0700,
+				Modified: &timestamp.Timestamp{
+					Seconds: 12,
+					Nanos:   0,
+				},
+				IsDir: false,
+			},
+			Stat: &fspb.FileStat{
+				Dev:     0,
```

Comment:
Some of these fields will not match after a restore, essentially the reporter is going to report every file and directory as not matching. What's the plan to handle those? Here are the ones that seem will not match (just from inspection):

- `Inode`: will be different on restore
- `Atime`, `Mtime`, `Ctime`: not all of them are captured in the snapshot, and also for directories Kopia records the most recent modification time of any of its children. Also, on restore, one of the times changes as soon as a metadata operation is performed, and if a change of owner or permissions is needed, the times are automatically reset.

These should match in most cases, but there are circumstances in which they may differ
- `Nlink`: in some cases this may be different, such as a file that has multiple hard link in the source directory. On restore, these will become multiple files.
- Uid and Gid: when the user used for restoring does not match the original one, and the restoring user has no privileges to change the ownership of the file(s).
- `Dev` and `Rdev`: in tests they should match, since the same device is used both as the source and the restore destination.
- `Blksize`: this depends on the source and target FS, again, in our setup this should not be an issue because the same FS (type, device, etc) will be used as source and destination. If `Blksize` changes, so would change the number of `Blocks`

Reply:
So here's the comment where the Reporter does its file comparison:
```
// diffFileStat compares the FileStat proto of two files and reports all relevant diffs as human readable strings.
// The following fields are ignored as they are not regarded as relevant in this context:
//   - atime
//   - inode, nlink, dev, rdev
//   - blksize, blocks
// The following fields are ignored as they are already part of diffFileInfo() check
// which is more guaranteed to be available (to avoid duplicate output):
//   - mode
//   - size
//   - mtime
```
So some of these fields are already being ignored.

I can think of two options here:
1. Use the Reporter for data verification
  - we can string parse to filter individual diffs that should be ignored but aren't (ex. look in the diff list for `"ctime: "`)
2. Write our own `Compare` functionality, evaluating the `fspb.Walk`s ourselves.
  - explicit control over the attributes we're interested in
  - don't need to string parse
  - wouldn't need the reporter wrapper anymore?

Reply:
> 
> 
> I think we should probably roll with the latter. The former won't work out of the box the way we want to use it, assuming we're going with our original plan of restoring to a new directory instead of on top of the source data.

Actually I can get around that problem by setting the walk file paths relative to the base path that I snap from / restore to

Reply:
Report outputs a series of diff strings highlighting anything changed.
e.g.
```
ctime: 2020-01-29 01:39:53 UTC => 2020-01-29 01:40:01 UTC
mtime: 2020-01-29 01:39:53 UTC => 2020-01-29 01:39:54 UTC
```
shouldn't be too hard to match fields we want to ignore. If we can rely on the existing reporter method, I don't have to copy and tweak a huge chunk of compare code. Can just wrap the report and filter out expected diffs.